ethdb/database.go

```diff
@@ -301,6 +303,15 @@ func (db *LDBDatabase) meter(refresh time.Duration) {
 				lastWriteDelay = time.Now()
 			}
 		}
+		// If leveldb's compaction has been going on for a long time, print a warning log here.
+		// If a warning that db is performing compaction has been displayed,
+		// any subsequent warnings will be withhold for 1 minute to don't overwhelm the user.
+		if paused && delayN-delaystats[0] == 0 && duration.Nanoseconds()-delaystats[1] == 0 &&
+			time.Now().After(lastWritePaused.Add(writeDelayWarningThrottler)) {
+			db.log.Warn("Database is performing the compaction, all write operations will be paused")
```

Comment:
I'd go with a shorter warning so it looks tidy among the other logs.

```
db.log.Warn("Database compacting, degraded performance")
```

Reply:
> @eqzip Hey, when geth receives a CTRL+C signal, it will try to exit.
> Unfortunately, your local node is performance a chain insert operation, while the db write operation is stuck by level db compaction. So geth will wait the chain insert operation and then exit gratefully.
> 
> So that is the reason why your geth node still prints warning log.

But I got to think that if I use Ctrl-c to exit then it should at least stop printing ,  it need a signal or context implementation when loop blocked .

Reply:
it may like sth in loop:
// to make a signal listen
select {
    case <- printSignal:
    log.Warn("xxx")
    case <- quitSignal:
    return
}